src/Evolution/Executables/GeneralizedHarmonic/EvolveGhBinaryBlackHole.hpp

```diff
@@ -264,38 +248,35 @@ struct EvolutionMetavars {
   void pup(PUP::er& /*p*/) {}
 
   template <::domain::ObjectLabel Horizon, typename Frame>
-  struct Ah : tt::ConformsTo<intrp::protocols::InterpolationTargetTag> {
-    static constexpr size_t index = static_cast<size_t>(Horizon);
-    using temporal_id = ::Tags::AhObservationTime<index>;
-    using vars_to_interpolate_to_target =
-        ::ah::vars_to_interpolate_to_target<volume_dim, Frame>;
-    using compute_vars_to_interpolate = ah::ComputeHorizonVolumeQuantities;
-    using tags_to_observe = ::ah::tags_for_observing<Frame>;
-    using surface_tags_to_observe = ::ah::surface_tags_for_observing;
-    using compute_items_on_source =
-        tmpl::list<::Tags::AhObservationTimeCompute<index>>;
-    using compute_items_on_target =
-        ::ah::compute_items_on_target<volume_dim, Frame>;
-    using compute_target_points = ah::TargetPoints::ApparentHorizon<Ah, Frame>;
-    using post_interpolation_callbacks =
-        tmpl::list<intrp::callbacks::FindApparentHorizon<Ah, Frame>>;
-    using horizon_find_failure_callbacks = tmpl::append<
-        tmpl::list<intrp::callbacks::IgnoreFailedApparentHorizon>,
+  struct Ah : tt::ConformsTo<ah::protocols::HorizonMetavars> {
+    static constexpr size_t index = 10 + static_cast<size_t>(Horizon);
+
+    using time_tag = ah::Tags::ObservationTime<index>;
+
+    using frame = Frame;
+
+    using horizon_find_callbacks = tmpl::append<
         tmpl::conditional_t<
             Horizon == ::domain::ObjectLabel::C,
-            tmpl::list<
-                intrp::callbacks::SendDependencyToObserverWriter<false, Ah>>,
-            tmpl::list<>>>;
-    using post_horizon_find_callbacks = tmpl::append<
+            tmpl::list<ah::callbacks::SendDependencyToObserverWriter<Ah, true>>,
+            tmpl::list<>>,
+        tmpl::list<ah::callbacks::ObserveFieldsOnHorizon<
+                       ::ah::surface_tags_for_observing, Ah>,
+                   ah::callbacks::ObserveTimeSeriesOnHorizon<
+                       ::ah::tags_for_observing<Frame>, Ah>>>;
+    using horizon_find_failure_callbacks = tmpl::append<
+        tmpl::list<ah::callbacks::FailedHorizonFind<Ah, true>>,
```

Comment:
Should we be ignoring failures for AhA and AhB?

Reply:
These are for observations only, so yes. If an observation horizon finder failed, we don't want to have to rerun the simulation if the control sys horizon finder is doing ok. However, since #6652, the observation horizon finder fails very rarely so it shouldn't be a big issue, but I still think we should ignore failed observation horizon finds.

Reply:
I think if the observations aren't working, we will have to rerun because we're missing output data, so we should fail quickly to not waste time.